Fix header error, update notifications

Achu-Worifung opened the pull request:
Fixed sidebar navigation error.
updated notifications
Create empty Profile and Settings page
added hover color to notification
replaced sign-in button with profile picture ( code for sign-in button was not deleted )

maishaSupritee reviewed (changes requested):
- Why are you using React.ComponentType<React.SVGProps<SVGSVGElement>> as the type for icons in types.ts file? Simply use React.ReactNode like I did for FAQData.
- Remove the title notifications for the notifications drop down.
![image](https://github.com/user-attachments/assets/5f286350-af04-435a-8c73-7f9e6455f4a4)
- The user button with the image i.e. the one that is displayed when someone is logged in should be an avatar component itself so we can reuse it everywhere in our app. You can use the avatar component from shadcn for this.

Achu-Worifung commented:
created the avatar component for the user profile
![image](https://github.com/user-attachments/assets/38cf30d1-fb99-45ae-9ef1-38395fd82a0c)
removed notification title from notification dropdown
![image](https://github.com/user-attachments/assets/5058a441-17b0-472c-a260-d464febd762a)
switched from React.ComponentType<React.SVGProps> to React.ReactNode
![image](https://github.com/user-attachments/assets/15d694aa-2f91-48e2-a1c9-14a04c38b177)

maishaSupritee reviewed (changes requested):
![image](https://github.com/user-attachments/assets/b477eaa5-c84b-49fe-b31c-ef795887ed47)

The padding on the top is way too less.
Also right now if i open the notifications bar i can immediately see the scroller inside of it, make it so that when I hover over the notifications only then I can see the scroller appear, regularly just clicking on notifications button shouldnt show the scroller as well. Look at how facebook's notifications drop down works.

![image](https://github.com/user-attachments/assets/b508037d-36a0-487b-be64-55a1fc8c2dc6)

The notifications title still exists in smaller screens. Remove. We don't need the cross button on the top right for the entire notifications panel in these smaller screens either, it should just close when we click anywhere outside of the notifications panel like our sidebar does.
